libpod/container.go

```diff
@@ -455,6 +452,11 @@ func newContainer(rspec *spec.Spec, lockDir string) (*Container, error) {
 	return ctr, nil
 }
 
+// SetSpec sets the container runtime spec.
+func (c *Container) SetSpec(rspec *spec.Spec) {
```

Comment:
Very much against this. We should not allow containers to be modified in a major way like this after creation.

I recommended earlier that we have a GetSpecFromImage() in libpod that will get a spec based on an image (set the command, etc). We can call that in podman to get a spec for the image, then pass the spec down into NewContainer without needing major modifications to the way we handle specs in the container itself.

Reply:
The big issue, I see is handling of the User flag. We need to have a container storage in place so that it can be mounted and then we can run a chroot to translate "user" into a UID/GID pair based on the /etc/password in the image.

With the latest code we can role back some of the changes of the spec earlier.

Reply:
We don't need the RW layer present to do the chroot/lookup, though - we're just readimg, not writing. So we can do this before the container is created by just mounting the image and chrooting into it. Libpod has the store, so this shouldn't be a big deal.

Reply:
Currently container/storage has no way of mounting an image without creating a layer, (I believe).  @nalind am I correct in that thinking?

Reply:
Maybe we should handle image users separately, as they're the only things that require a mounted container. We can have a DefaultSpecFromImage() that takes an image and parses everything except the user into a usable default config we can use in podman create. Then we can have a separate WithImageUser() that will tell libpod to handle the chroot'd passwd/group parsing when the container is started.

Reply:
I pushed a new version with a WithUser parameter to setup the correct UID/GID for the container.  I am not sure if we need to store these in the sql database or not.  Removed the code to setup the runtimespec separately.